### ISSUE TITLE
I updated to Macosx Sequoia and now hooks to gotify fail

### ISSUE THREAD

**Issue by Bredyit** [garethgeorge/backrest#498]
Errors with run Hook... 
I no longer receive messages sent to Gotify, and I have errors like this:
error: send gotify message: send request http://10.21.0.106:80/message?token=XbKJH8Py.fVUz-y: Post "http://10.21.0.106:80/message?token=XbKJH8Py.fVUz-y": dial tcp 10.21.0.106:80: connect: no route to host

note : backup run correctly, i have given full disk and network permission to backrest, but I have no idea what to do regarding failed messages to Gotify

**Platform Info**
Platform Info

    MacOS 15.0 Sequoia Darwin arm64
    Backrest 0.14.0

**Comment by garethgeorge** [comment]
Hey, possibly silly question -- but did you restart Backrest after granting it full access? I don't have access to a macOS system to test on for the next few weeks and am seeing a number of similar reports after the upgrade.

**Comment by Bredyit** [comment]
Yes, I restart my mac every day, but it hasn't improved my situation. However, I notice that I installed restic via homebrew, while for backrest instead I downloaded the compressed file and then I installed it with the appropriate instructions. I'm curious to see when we will have the first update of macosx sequoia, if the problem remains...

**Comment by garethgeorge** [comment]
Annoying, following this issue in a few other threads -- macOS tends to make it difficult to understand these permission changes as they don't explicitly document what an app needs to do to gain trust / ensure it stays allowlisted.

It sounds like Backrest still isn't on the network allowlist from your symptoms.

**Comment by unfixa1** [comment]
Homebrew is not compatible with Mac and has many bugs.
It is recommended that you use tools such as [servbay](servbay.com) to check the development environment and try redeploying.

**Comment by Bredyit** [comment]
I uninstalled restic from homebrew and reinstalled it via the binary, good thing that the mac noticed the launch of restic and asked me to give consent to use it.
I think I'll also reinstall backrest, because my problem remains related to my hook to Gotify, in fact the saves are executed, only the hooks are my problem.
As soon as I find the time, first I save the configurations and then I reinstall the latest version available, but it will take me some time, I predict.

**Comment by garethgeorge** [comment]
Appreciate your updating on what you're trying. To add context for other users looking at this issue (and gathered from related Sequoia bugs)

 * Backrest installed via homebrew may change it's path on each upgrade which can invalidate previously allowlisted permissions (even if appearing to be allowlisted in the UI). If aiming for full disk access OR local network access, please download backrest as a tar.gz from releases and install OR update using install.sh . Ensure that you remove your homebrew install.
 * Backrest manages it's own version of restic , by default this is in `~/.local/share/backrest/restic-0.17.1` (or some such version). Allowlisted permissions e.g. network access OR disk access should be granted to the backrest process (installed by install.sh to /usr/local/bin/backrest). If that doesn't work, you may also try allowlisting the restic binary backrest manages, but that isn't expected to be required.

**Comment by Bredyit** [comment]
Hi, it's been a while and I've updated the mac mini M2 to macosx sequoia 15.1 and Backrest to version 1.6.1 - Despite having granted disk access authority every time the OS asked me for it (later granted access to the full disk), I also added Backrest to the "Local Network" section but I find failed saves most of the time,  not just the old problem of Hook in Gotify.
However, I could notice today that the saving that failed, was resolved when I set the machine in dhcp and to my manual save request, I received from Backrest the message "API failure" and then "Api reconneted" and then the application from here started working again.
I hope this can help to solve the problem described.
Thank you
NB= For now I left the machine in DHCP bound to a specific address (defined at the router level), so I can see how it behaves in the next few days.  And to think that before Sequoia everything worked.

**Comment by Bredyit** [comment]
After a long time, I solved the problem by deleting the destination shared folder located in "/Volumes" and creating a new one in my "home" folder. To do this, I created a bash script that unmounts the folder if it exists and remounts it from scratch. I placed the script in /usr/local/bin/Create the script**

Create a file called, for example:

`/usr/local/bin/reset_smb.sh`

Contents:

bash

```
#!/bin/bash

SHARE_PATH="/Volumes/SHARENAME"
SERVER="//user:password@SERVER_IP/SHARENAME"

# Force unmount if it exists
if mount | grep -q "$SHARE_PATH"; then
umount -f "$SHARE_PATH"
fi

# Wait a second
sleep 1

# Recreate the mount directory if it doesn't exist
mkdir -p "$SHARE_PATH"

# Remount the share
mount_smbfs "$SERVER" "$SHARE_PATH"
```

Then make it executable:

bash

```
sudo chmod +x /usr/local/bin/reset_smb.sh
```

Automate it with LaunchAgent**

Create the file:

`~/Library/LaunchAgents/com.reset.smb.plist`

Contents:

xml

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict> 
<key>Label</key> 
<string>com.reset.smb</string> 

<key>ProgramArguments</key> 
<array> 
<string>/usr/local/bin/reset_smb.sh</string> 
</array> 

<key>RunAtLoad</key> 
<true/>
</dict>
</plist>
```

Activate it:

bash

```
launchctl load ~/Library/LaunchAgents/com.reset.smb.plist
```

Now I solved it

tanks

**Comment by Bredyit** [comment]
Now I solved it, previus msg is solution for me.
